iOS/MSQASignInSwift/MSQASignInButton.swift

```diff
@@ -0,0 +1,157 @@
+
+  /// Creates an instance of Microsoft Quick Auth sign in button.
+  /// - parameter viewModel: An instance of `MSQASignInButtonViewModel` with customization about the button's type, theme, size, etc.
+  ///     Defaults to `MSQASignInButtonViewModel` with its standard defaults.
+  /// - parameter action: A closure to be executed on button press event.
+  ///     Defaults to no-op.
+  public init(
+    viewModel: MSQASignInButtonViewModel = MSQASignInButtonViewModel(),
+    action: @escaping () -> Void = {}
+  ) {
+    self.viewModel = viewModel
+    self.action = action
+  }
+
+  /// A convenience initializer to create a Microsoft Quick Auth sign in button with type, theme, size, etc.
+  /// - parameter type: The `MSQASignInButtonType` to use.
+  ///     Defaults to `.standard`.
+  /// - parameter theme: The `MSQASignInButtonTheme` to use.
+  ///     Defaults to `.light`.
+  /// - parameter size: The `MSQASignInButtonSize` to use.
+  ///     Defaults to `.large`.
+  /// - parameter text: The `MSQASignInButtonText` to use.
+  ///     Defaults to `.signInWith`.
+  /// - parameter shape: The `MSQASignInButtonShape` to use.
+  ///     Defaults to `.rectangular`.
+  /// - parameter layout: The `MSQASignInButtonLayout` to use.
+  ///     Defaults to `.logoTextLeft`.
+  /// - parameter state: The `MSQASignInButtonState` to use.
+  ///     Defaults to `.normal`.
+  /// - parameter action: A closure to be executed on button press event.
+  ///     Defaults to no-op.
+  public init(
+    type: MSQASignInButtonType = .standard,
+    theme: MSQASignInButtonTheme = .light,
+    size: MSQASignInButtonSize = .large,
+    text: MSQASignInButtonText = .signInWith,
+    shape: MSQASignInButtonShape = .rectangular,
+    layout: MSQASignInButtonLayout = .logoTextLeft,
+    state: MSQASignInButtonState = .normal,
+    action: @escaping () -> Void = {}
+  ) {
+    let viewModel = MSQASignInButtonViewModel(
+      type: type, theme: theme, size: size, text: text, shape: shape, layout: layout, state: state)
+    self.init(viewModel: viewModel, action: action)
+  }
+
+  public var body: some View {
+    Button(action: action) {
+      switch viewModel.type {
+      case .standard:
+        switch viewModel.layout {
+        case .logoTextLeft: logoTextLeadingView
+        case .logoLeftTextCenter: logoLeadingTextCenterView
+        case .logoTextCenter: logoTextCenterView
+        }
+      case .icon: iconView
+      }
+    }
+    .font(viewModel.buttonStyle.font)
+    .buttonStyle(viewModel.buttonStyle)
+    .simultaneousGesture(
+      DragGesture(minimumDistance: 0)
+        .onChanged({ _ in viewModel.state = .pressed })
+        .onEnded({ _ in viewModel.state = .normal })
+    )
+  }
+
+  private var logoTextLeadingView: some View {
+    return HStack(alignment: .center, spacing: 0) {
+      iconView
+        .padding([.leading, .trailing], MSQASignInButtonElementPadding)
+      Text(MSQASignInButtonHelper.localizedString(forText: viewModel.text))
```

Comment:
Please add the accessibility support for the localized text, `accessibilityLabel`?

Reply:
Please check with Arabic, whose layout is "right to left", to see if the layout is correct. Unlike UIKit, where we need to calculate the text's bound, I'm not sure whether `Text` in SwiftUI can handle the case of RTL automatically.
https://github.com/microsoft/quick-authentication-mobile/blob/c3e9dd3510b7d6f6f3b157bb21a773b54e7e512c/iOS/MSQASignIn/src/MSQASignInButton.m#L587-L589